packages/core/src/util/log.ts

```
@@ -1,12 +1,64 @@
+
+    const actualParams = (optionalParams ?? []).map(p => (typeof p === "function" ? p() : p));
+    const structuredMsg = {
+      ...msg,
+      ...actualParams,
+      timestamp: new Date().toISOString(),
+      requestId: reqId,
+      source,
+    };
+
+    writeToFile(structuredMsg);
+    return console.log(structuredMsg);
```

Biome (1.9.4)

[error] The function should not return a value because its return type is void.

The function is here:

'void' signals the absence of value. The returned value is likely to be ignored by the caller.

(lint/correctness/noVoidTypeReturn)